spacy/pipeline/entity_linker.py

```diff
@@ -453,126 +445,129 @@ def predict(self, docs: Iterable[Doc]) -> ActivationsT:
             }
         if isinstance(docs, Doc):
             docs = [docs]
-        for doc in docs:
+
+        docs = list(docs)
```

Comment:
Why do we need to call `list()` on the `Iterable[doc]`? If we can avoid it, we don't end up having all the `docs` in memory, but we can stream them instead.

Reply:
That's true. The issue is that currently we consume `docs` in two places: to generate `all_ent_cands` and in the loop over all documents, hence the list conversion. It should be possible to rewrite this so we don't need to consume docs twice, I'll look at that.

Reply:
Hm, so there isn't an easy way to avoid consuming `docs` twice because we `get_candidates()` reads entities from the docs and main loop needs access to neighboring sentences in the `incl_context` block. 

One possible alternative is to have `get_candidates()` return the `Doc` instance together with the candidates - this maintains both a low memory profile and allows performant batching, but it's very clunky. Open to other suggestions.

Reply:
We could `tee()` it.